Libraries/plist/Tools/PlistBuddy.cpp

```diff
@@ -0,0 +1,572 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include <libutil/Options.h>
+#include <libutil/DefaultFilesystem.h>
+#include <plist/Array.h>
+#include <plist/Boolean.h>
+#include <plist/Data.h>
+#include <plist/Date.h>
+#include <plist/Dictionary.h>
+#include <plist/Integer.h>
+#include <plist/Real.h>
+#include <plist/String.h>
+#include <plist/Object.h>
+#include <plist/Format/Any.h>
+#include <plist/Format/ASCII.h>
+#include <plist/Format/Binary.h>
+#include <plist/Format/Encoding.h>
+#include <plist/Format/XML.h>
+
+#include <histedit.h>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <queue>
+
+typedef struct {
+    std::unique_ptr<plist::Object> object;
+} RootObjectContainer;
+
+class Options {
+public:
+
+private:
+    bool                     _help;
+    bool                     _xml;
+
+private:
+    std::string              _command;
+
+private:
+    std::string              _input;
+
+public:
+    Options();
+    ~Options();
+
+public:
+    bool help() const
+    { return _help; }
+    bool xml() const
+    { return _xml; }
+
+public:
+    std::string command() const
+    { return _command; }
+
+public:
+    std::string input() const
+    { return _input; }
+
+private:
+    friend class libutil::Options;
+    std::pair<bool, std::string>
+    parseArgument(std::vector<std::string> const &args, std::vector<std::string>::const_iterator *it);
+
+};
+
+Options::
+Options() :
+    _help   (false),
+    _xml    (false),
+    _command("")
+{
+}
+
+Options::
+~Options()
+{
+}
+
+std::pair<bool, std::string> Options::
+parseArgument(std::vector<std::string> const &args, std::vector<std::string>::const_iterator *it)
```

Comment:
(`libutil::Options` was changed to use `ext::optional` recently, so this needs to be updated; see `plutil.cpp` for an example.)